src/codeflare_sdk/cluster/cluster.py

```diff
@@ -547,6 +547,20 @@ def _component_resources_up(
         if self.config.write_to_file:
             with open(self.app_wrapper_yaml) as f:
                 yamls = yaml.load_all(f, Loader=yaml.FullLoader)
+                for resource in yamls:
+                    enable_ingress = (
+                        resource.get("spec", {})
+                        .get("headGroupSpec", {})
+                        .get("enableIngress")
+                    )
+                    if resource["kind"] == "RayCluster" and enable_ingress is not False:
+                        name = resource["metadata"]["name"]
+                        print(
+                            f"Forbidden: RayCluster '{name}' has 'enableIngress' set to 'True' or is unset."
+                        )
+                        return
+                f.seek(0)  # Reset file pointer to the beginning
+                yamls = yaml.load_all(f, Loader=yaml.FullLoader)  # Reload the YAMLs
```

Comment:
Why are we loading the yaml here again? I think in the case where this error occurs this function would be run again anyway in order to apply the updated yaml.

Reply:
Previously, the `yaml.load_all` would use a generator which can only be iterated over once, for this reason reloading the YAML was necessary to then be used in the RayCluster creation (probably not the best approach).

Thanks for pointing this out, I now opted to convert it to a list instead in the latest commit, this way we can iterate through it as needed without having to reload the YAML. - Overall behaviour is the same as before.

Some refs: 
https://stackoverflow.com/questions/42802058/pyyaml-load-all-returns-generator-rather-than-a-dict 
https://github.com/yaml/pyyaml/issues/762